tests.unit/test_utility_functions.py

```diff
@@ -270,12 +270,36 @@ def test_export_c_type_array_32_bytes_and_line_cont(self):
         self.assertEqual(expected_output, output.getvalue())
 
     def test_export_c_type_array_empty(self):
-        """Ensure exception is raised if the array is length 0"""
+        """Ensure exception is raised if the array is length 0."""
         binary_data = io.BytesIO(b"")
         output = io.StringIO()
 
         with self.assertRaises(ValueError):
             utilities.export_c_type_array(binary_data, "TestVariable", output)
 
+class TestRemoveTree:
+    """Tests the RemoveTree function."""
+    @pytest.mark.skipif(not sys.platform.startswith('win'), reason="Long Paths are only an issue on Windows")
+    def test_long_path_remove_tree(self, tmp_path):
+        """Tests RemoveTree's ability to remove a directory on a Windows System with LongPaths Disabled."""
+        import winreg
+        sub_key = r"SYSTEM\CurrentControlSet\Control\FileSystem"
+        value_name = "LongPathsEnabled"
+
+        key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, sub_key)
+        value, _ = winreg.QueryValueEx(key, value_name)
+        if value == 1:
+            pytest.skip(r"Long paths are enabled. To run the test, disable long paths with the registry key located at: HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystemLongPathsEnabled")  # noqa: E501
+
+        long_path = str(tmp_path / ("a" * 250)) # A single folder cannot be longer than 260 characters.
```

Comment:
I assume `tmp_path` is passed in a way that it's guaranteed to be to provide the additional characters necessary to reach 260?

Reply:
On a windows system, pytest creates a tmp_path at `C:/Users/<user>/AppData/Local/Temp/`, however I do have an assert on L295 that ensures the overall path is greater than 260.